src/engine.rs

```diff
@@ -35,7 +35,15 @@ pub struct Engine {
 struct EndPointVoices {
     format: Format,
     next_id: AtomicUsize,
-    pending_sounds: Mutex<Vec<(usize, QueueIterator)>>,
+    pending_sounds: Mutex<Vec<HandleHandle>>,
+}
+
+// Handle to a handle I guess.
+// The tuple type was getting a bit unwieldy so I added this.
+struct HandleHandle {
+    handle_id: usize,
+    queue_iterator: QueueIterator,
+    dead: Arc<Mutex<bool>>,
```

Comment:
`dead` should be a `Arc<AtomicBool>` instead of a `Mutex<bool>`.
You can use `Ordering::Relaxed` if you're not familiar with the ordering stuff.